Sources/SwiftProtobuf/BinaryEncodingVisitor.swift

```diff
@@ -338,7 +370,9 @@ extension BinaryEncodingVisitor {
       let length = try value.serializedDataSize()
       encoder.putVarInt(value: length)
       // Create the sub encoder after writing the length.
-      var subVisitor = BinaryEncodingVisitor(forWritingInto: encoder.pointer)
+      var subVisitor = BinaryEncodingVisitor(
+        forWritingInto: encoder.pointer, options: BinaryEncodingOptions()
```

Comment:
This seems wrong, don't you need the correct options or else the submessage here won't be able to honor the requested options?

Reply:
Yep, not sure why I missed this originally - updated.